### PR TITLE
chore: upgrade template to 0.4.0

### PR DESCRIPTION
**Pull request by hardfist**
## Summary
upgrade template to Rspack 0.4.0

## Test Plan
we do need for eco-ci

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by h-a-n-a — DISMISSED** [review]

**Comment by hardfist** [comment]
!canary

File: packages/create-rspack/template-react-ts/rspack.config.js [hunk 1]
```diff
@@ -12,12 +14,44 @@ module.exports = {
...
+								transform: {
+									react: {
+										runtime: "automatic",
+										development: isDev,
+										refresh: isDev
+									}
+								}
+							},
+							env: {
+								targets: "Chrome >= 48"
```
Comment (chenjiahan) [inline packages/create-rspack/template-react-ts/rspack.config.js:39]: How about a modern browserslist by default

```ts
[
  'chrome >= 87',
  'edge >= 88',
  'firefox >= 78',
  'safari >= 14',
];
```

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -8987,6 +9123,14 @@ packages:
     dev: true
     optional: true
 
+  /@rspack/binding-linux-arm64-gnu@0.3.14:
```
Comment (chenjiahan) [inline pnpm-lock.yaml:9126]: Why is Rspack 0.3.14 added to lock file

Comment (hardfist) [reply pnpm-lock.yaml:9126]: good points, forget to update vue-template version

**Comment by hardfist** [comment]
!canary

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**